### PR TITLE
Adds routing

### PR DESCRIPTION
**Pull request by jlhogan**
Figured this would have to happen eventually, and doesn't require designs! 
- adds react-router with some basic routing and a placeholder Home and ResultDetail page
- adds go back to ResultDetail and persists filters from URL params to make filtered data sets share-able via links, and easily return-to-able
- pulls out search logic into Search page
- Adds search nav link to header

### CODE REVIEW

File: src/util.tsx [hunk 1]
```diff
@@ -30,17 +30,14 @@ export type CareEntity = {
   date_geocoded: string;
   data_source: string;
   dataset_name: string;
-};
-
-export type CareResult = CareEntity & {
-  distance: number;
```
Comment (jlhogan) [inline src/util.tsx:36]: I was having trouble keeping track of CareEntity vs CareResult and felt like it was simpler to just add optional `distance` field to the CareEntity type. Other thoughts/ objections/ideas about this?

File: src/components/MilesAway.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
-export default function MilesAway({ meters }: { meters: number }) {
-  const miles = meters * 0.000621371192;
+export default function MilesAway({ meters }: { meters?: number }) {
+  const miles = meters ? (meters * 0.000621371192).toFixed(1) : '??';
```
Comment (jlhogan) [inline src/components/MilesAway.tsx:2]: accounting for `distance` being optional on the single `CareEntity` type

File: src/App.tsx [hunk 3]
```diff
@@ -1,139 +1,29 @@
...
-                  ))}
-            </CardGroup>
-          </>
-        )}
-      </GridContainer>
+      <Routes>
+        <Route path="/" element={<Layout />}>
+          <Route index element={<Home />}/>
+          <Route path="search" element={<Search />} />
+          <Route path="result/:resultId" element={<ResultDetail />} />
```
[the remark this answers is not in the document]
Comment (cathydeng) [reply src/App.tsx:22]: `provider`? `location`? `detail`?

File: src/util.tsx (hunk 1 above)
Comment (cathydeng) [reply src/util.tsx:36]: the distinction i was going for is that `CareEntity` is just the data, whereas `CareResult` (maybe better named as `CareSearchResult`) is the entity data + extra data on how it relates to the search. was thinking we may want some other things on this, e.g. order in a numbered list of search results, possibly also metadata about how strong a match is for a given filter (e.g. there are some filters where it's unclear what to do with missing data, not sure if we'll eventually want some representation of match strength)

**Review by cathydeng — APPROVED** [review]
nice, thx!! lmk if it's helpful to chat live re: how we want to structure search metadata

**Comment by jlhogan** [comment]
` lmk if it's helpful to chat live re: how we want to structure search metadata` <-- i think this will be easier to do once we have more insight into what metadata will be useful/necessary, so i think we can punt on that til now and reintroduce the separate types when it become necessary

**Pull request merged**